foursight_core/app_utils.py

```diff
@@ -273,7 +273,7 @@ def set_user_record(self, email: str, record: Optional[dict], error: Optional[st
         user_record["error"] = error
         user_record["exception"] = exception
 
-    def get_portal_url(self, env_name: str) -> Optional[str]:
+    def get_portal_url(self, env_name: str, raise_exception: bool = False) -> Optional[str]:
```

Comment:
On line 280, below, which I am unable to point to directly in this review, I'd prefer to see a continuation line done with parentheses. Those backslashes are ugly and have to get manually removed if the line gets merged whereas parens are semantically valid either way.

Reply:
Ah, yeah, I know we don't like those, I fall back to it sometime, thanks, will fix.